windows/src/developer/TIKE/compile/MergeKeyboardInfo.pas

```diff
@@ -289,21 +291,57 @@ function TMergeKeyboardInfo.LoadKMPFile: Boolean;
       end;
 
       FKMXTempFile := TTempFileManager.Get('.kmx');
+      FJSTempFile := TTempFileManager.Get('.js');
+
       try
-        for i := 0 to High(Zip.FileNames) do
+        if FMergingValidateIds then
         begin
-          if SameText(ExtractFileExt(Zip.FileNames[i]), '.kmx') then
+          for i := 0 to FKMPInfFile.Keyboards.Count - 1 do
           begin
-            SetLength(FKMXFiles, Length(FKMXFiles)+1);
-
-            SaveMemberToFile(Zip.FileNames[i], FKMXTempFile.Name);
+            for j := 0 to High(Zip.FileNames) do
+            begin
+              // Add the KMX to FPackageKMXFileinfos
+              if SameText(Zip.FileName[j], FKMPInfFile.Keyboards[i].ID + '.kmx') then
+              begin
+                SetLength(FPackageKMXFileInfos, Length(FPackageKMXFileInfos)+1);
+                SaveMemberToFile(Zip.FileNames[j], FKMXTempFile.Name);
+                FPackageKMXFileInfos[High(FPackageKMXFileInfos)].Filename := Zip.FileNames[j];
+                GetKeyboardInfo(FKMXTempFile.Name, False, FPackageKMXFileInfos[High(FPackageKMXFileInfos)].Info, False);
+              end;
+
+              // Add the JS to FPackageJSFileInfos
+              if SameText(Zip.FileNames[j], FKMPInfFile.Keyboards[i].ID + '.js') then
+              begin
+                SetLength(FPackageJSFileInfos, Length(FPackageJSFileInfos)+1);
+                SaveMemberToFile(Zip.FileNames[j], FJSTempFile.Name);
+                FPackageJSFileInfos[High(FPackageJSFileInfos)].Filename := Zip.FileNames[j];
+
+                // For now, apply JS keyboard to all web and mobile targets
+                // Not using GetKeyboardInfo because that only handles kmx files
+                FPackageJSFileInfos[High(FPackageJSFileInfos)].Info.Targets := 'web mobile';
+              end;
+            end;
+          end;
+        end
 
-            FKMXFiles[High(FKMXFiles)].Filename := Zip.FileNames[i];
-            GetKeyboardInfo(FKMXTempFile.Name, False, FKMXFiles[High(FKMXFiles)].Info, False);
+        // Handle legacy keyboards which may not have keyboards defined in kmp.inf
```

Comment:
This comment should note that a package without a Keyboards section (or empty Keyboards) section never will include any js keyboards.